src/Microsoft.AspNetCore.Hosting/Internal/StartupLoader.cs

```diff
@@ -11,9 +12,45 @@ namespace Microsoft.AspNetCore.Hosting.Internal
 {
     public class StartupLoader
     {
+        /// <summary>
```

Comment:
I can make it a regular comment. I just wanted to document the process as its non trivial and gives a head start to anyone that has to touch this code in the future.